lib/commanded/commands/router.ex

```diff
@@ -288,48 +290,19 @@ defmodule Commanded.Commands.Router do
 
   """
   defmacro identify(aggregate_module, opts) do
-    quote location: :keep, bind_quoted: [aggregate_module: aggregate_module, opts: opts] do
-      case Map.get(@registered_identities, aggregate_module) do
-        nil ->
-          by =
-            case Keyword.get(opts, :by) do
-              nil ->
-                raise "#{inspect(aggregate_module)} aggregate identity is missing the `by` option"
-
-              by when is_atom(by) ->
-                by
+    aggregate_module =
+      Macro.expand_literals(aggregate_module, %{__CALLER__ | function: {:dummy, 1}})
 
-              by when is_function(by, 1) ->
-                by
-
-              invalid ->
-                raise "#{inspect(aggregate_module)} aggregate identity has an invalid `by` option: #{inspect(invalid)}"
-            end
-
-          prefix =
-            case Keyword.get(opts, :prefix) do
-              nil ->
-                nil
+    {opts, _bindings} = Code.eval_quoted(opts, [], __CALLER__)
```

Comment:
Hex docs for the [`Code.eval_quoted/3` function](https://hexdocs.pm/elixir/Code.html#eval_quoted/3) has the following warning which seems to be applicable here as the usage is within a macro.

> **Warning**: Calling this function inside a macro is considered bad practice as it will attempt to evaluate runtime values at compile time. Macro arguments are typically transformed by unquoting them into the returned quoted expressions (instead of evaluated).

I don't know what this code is supposed to be doing or why it is needed. The macros in this module are already difficult enough to understand without this added complexity. 

Can you help me (and other contributors to this code) to understand what is going on.

Reply:
Well spotted. I think I managed to rewrite the macro to avoid using `eval_quoted`.
It was needed to properly handle the case where a function-reference is passed to the `by` option.